src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -668,14 +668,22 @@ private void LoadAssemblies(IEnumerable<string> assemblies)
         /// </summary>
         private static IEnumerable<PortableExecutableReference> InitDefaultRefAssemblies()
         {
-            // netcoreapp3.0 currently comes with 148 reference assemblies (maybe more in future), so we use a capacity of '150'.
+            // PowerShell is distributed with 149 reference assemblies, so we use a capacity of '150'.
             var defaultRefAssemblies = new List<PortableExecutableReference>(150);
 
             foreach (string file in Directory.EnumerateFiles(s_netcoreAppRefFolder, "*.dll", SearchOption.TopDirectoryOnly))
             {
                 defaultRefAssemblies.Add(MetadataReference.CreateFromFile(file));
             }
+
+            // Add System.Management.Automation.dll
             defaultRefAssemblies.Add(MetadataReference.CreateFromFile(typeof(PSObject).Assembly.Location));
+
+            // We want to avoid reallocating the internal array, so we assert if the list capacity has increased.
+            System.Diagnostics.Debug.Assert(
```

Comment:
It seems we have custom Debug.Assert in PowerShell repo.

Reply:
I'm not sure what changes to make, can you make a suggestion?

Perhaps [`PSTraceSource`](https://docs.microsoft.com/dotnet/api/system.management.automation.pstracesource)?

Reply:
System.Management.Automation.Diagnostics.Assert